web/ui/react-app/src/pages/Flags.test.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import { mount, shallow, ReactWrapper } from 'enzyme';
+import { act } from 'react-dom/test-utils';
+import Flags from './Flags';
+import { Alert, Table } from 'reactstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+
+const sampleAlertmanagerResponse: {
+  status: string;
+  data: { [key: string]: string };
+} = {
+  status: 'success',
+  data: {
+    'alertmanager.notification-queue-capacity': '10000',
+    'alertmanager.timeout': '10s',
+    'config.file': './documentation/examples/prometheus.yml',
+    'log.format': 'logfmt',
+    'log.level': 'info',
+    'query.lookback-delta': '5m',
+    'query.max-concurrency': '20',
+    'query.max-samples': '50000000',
+    'query.timeout': '2m',
+    'rules.alert.for-grace-period': '10m',
+    'rules.alert.for-outage-tolerance': '1h',
+    'rules.alert.resend-delay': '1m',
+    'storage.remote.flush-deadline': '1m',
+    'storage.remote.read-concurrent-limit': '10',
+    'storage.remote.read-max-bytes-in-frame': '1048576',
+    'storage.remote.read-sample-limit': '50000000',
+    'storage.tsdb.allow-overlapping-blocks': 'false',
+    'storage.tsdb.max-block-duration': '36h',
+    'storage.tsdb.min-block-duration': '2h',
+    'storage.tsdb.no-lockfile': 'false',
+    'storage.tsdb.path': 'data/',
+    'storage.tsdb.retention': '0s',
+    'storage.tsdb.retention.size': '0B',
+    'storage.tsdb.retention.time': '0s',
+    'storage.tsdb.wal-compression': 'false',
+    'storage.tsdb.wal-segment-size': '0B',
+    'web.console.libraries': 'console_libraries',
+    'web.console.templates': 'consoles',
+    'web.cors.origin': '.*',
+    'web.enable-admin-api': 'false',
+    'web.enable-lifecycle': 'false',
+    'web.external-url': '',
+    'web.listen-address': '0.0.0.0:9090',
+    'web.max-connections': '512',
+    'web.page-title': 'Prometheus Time Series Collection and Processing Server',
+    'web.read-timeout': '5m',
+    'web.route-prefix': '/',
+    'web.user-assets': '',
+  },
+};
+
+describe('Flags', () => {
+  beforeEach(() => {
+    fetch.resetMocks();
+  });
+
+  describe('before data is returned', () => {
+    it('renders a spinner', () => {
+      const flags = shallow(<Flags />);
+      const icon = flags.find(FontAwesomeIcon);
+      expect(icon.prop('icon')).toEqual(faSpinner);
+      expect(icon.prop('spin')).toEqual(true);
+    });
+  });
+
+  describe('when data is returned', () => {
+    it('renders a table', async () => {
+      const mock = fetch.mockResponse(JSON.stringify(sampleAlertmanagerResponse));
```

Comment:
I think this is pretty cool. It seems slightly magical, but I like that your Flag component doesn't have to cater to the test code.